src/main/java/org/owasp/wrongsecrets/oauth/TokenController.java

```diff
@@ -31,9 +31,16 @@ public TokenController(@Value("${challengedockermtpath}") String dockerMountPath
     @Operation(summary = "Endpoint for interaction at challenge 16")
     @PostMapping(path = "/token", consumes = {MediaType.APPLICATION_FORM_URLENCODED_VALUE})
     public ResponseEntity<?> clientCredentialToken(TokenRequest tokenRequest) {
-        if ("client_credentials".equals(tokenRequest.grant_type())
-            && "WRONGSECRET_CLIENT_ID".equals(tokenRequest.client_id())
-            && getActualData().equals(tokenRequest.client_secret())) {
+
+        String grantType = tokenRequest.grant_type();
+        boolean isCredentialsMatching = "client_credentials".equals(grantType);
+
+        String clientId = tokenRequest.client_id();
+        String clientSecret = tokenRequest.client_secret();
+        boolean isClientIdMatching = "WRONGSECRET_CLIENT_ID".equals(clientId);
+        boolean isClientSecretMatching = getActualData().equals(clientSecret);
```

Comment:
i am not seeing the advantage of this for longer term maintenance, sorry. I am fine if you want to extract the if towards a (series) of methods that still embed null-safety , but for me it takes me a few reads to now see what is happening, whilst it made more sense to me earlier.